### PR TITLE
Added password checker in ruby

### PR DESCRIPTION
**Pull request by lsfernandes92**
Added password strength checker in ruby for String algorithms.
The folder: "code/string-algorithms/password_strength_checker".

### CODE REVIEW

File: code/string-algorithms/password_strength_checker/pw_checker.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  has_number = pattern_numbers =~ password
+  has_upcase_letter = pattern_upcase =~ password
+  has_downcase_letter = pattern_downcase =~ password
+  has_especial_char = pattern_special_char =~ password
+  has_min_length = password.size >= 8
+
+  is_strong_pass = (has_number != nil) && (has_min_length != false) && (has_upcase_letter != nil) &&
+   (has_downcase_letter != nil) && (has_especial_char != nil)
+
+  puts "The password doesn't the minimum length" if (has_min_length == false)
```
Comment (arnavb) [inline code/string-algorithms/password_strength_checker/pw_checker.rb]: Fix grammar here.

File: code/string-algorithms/password_strength_checker/pw_checker.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  puts "The password doesn't have a number digit" if (has_number == nil)
+  puts "The password doesn't have any upper case letter" if (has_upcase_letter == nil)
+  puts "The password doesn't have any down case letter" if (has_downcase_letter == nil)
```
Comment (arnavb) [inline code/string-algorithms/password_strength_checker/pw_checker.rb]: down case => lowercase (as one word). 😝

File: code/string-algorithms/password_strength_checker/pw_checker.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  has_downcase_letter = pattern_downcase =~ password
+  has_especial_char = pattern_special_char =~ password
+  has_min_length = password.size >= 8
+
+  is_strong_pass = (has_number != nil) && (has_min_length != false) && (has_upcase_letter != nil) &&
+   (has_downcase_letter != nil) && (has_especial_char != nil)
+
+  puts "The password doesn't the minimum length" if (has_min_length == false)
+  puts "The password doesn't have a number digit" if (has_number == nil)
+  puts "The password doesn't have any upper case letter" if (has_upcase_letter == nil)
```
Comment (arnavb) [inline code/string-algorithms/password_strength_checker/pw_checker.rb]: uppercase is one word.

**Comment by lsfernandes92** [comment]
Wow what a shame Lucas hahahahaha. Well... I sent the changes resquested, see if it's alright. 👍

**Pull request closed without merging**